gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -235,13 +245,23 @@ fluid.defaults("gpii.flowManager.local", {
             funcName: "gpii.flowManager.local.mountWebSocketsSettingsHandler",
             args: ["{webSocketsSettingsHandlerComponent}"]
         },
-
+        "onCreate.loadSolutionsFromRepository": {
```

Comment:
It might be better to have `loadSolutionsFromRepository`(reading the SR from Github) as part of `SolutionsRegistryDataSource` so it's transparent to outsiders including flowManager.

What `SolutionsRegistryDataSource` does it:
1. fetch the SR for the current platform from Github;
2. If 1 fails, use the SR read from the universal /testData/solutions directory;
3. If 2 fails again, call `fluid.fail` and exit.

All above happens at the "onCreate" event of `SolutionsRegistryDataSource`. When the process completes and SR is ready, fires an event `SolutionRegistryReady` that gets escalated to `flowManager` and becomes one of the events triggering the aggregate event `flowManagerReady`.

Reply:
This works for the `SRDS` used by the `LFM`.  But, the `CBFM` also has an `SRDS` sub-component.  In this case, the platform that the `CBFM` is running on is irrelevant.  The relevant platform is the one that the Morphic client is running on, and it communicates that to the `CBFM` via the `DeviceReporter`.  From this perspective, the `CBFM/SRDS` has to have access to all of the platforms' solutions registries since it can't know the client's platform.  Note that that is how the original `SRDS` worked, and I've kept that functionality for the `CBFM` case.

Even so, you mentioned that you found no use of the `SolutionsRegistryDataSource.getImpl()` function where the platform was passed in as one of its arguments.  I'm mystified, and am looking closer -- how does the client inform the CBFM what platform it (the client) is running on?